src/community.c

```diff
@@ -3567,3 +3567,1094 @@ int igraph_i_compare_communities_rand(const igraph_vector_t *v1,
+ * \param membership The membership vector. This is both used as the initial
+ *    membership from which optimisation starts and is updated in place. It
+ *    must hence be properly initialized. When finding clusters from scratch it
+ *    is typically started using a singleton clustering. This can be achieved
+ *    using \c igraph_vector_init_seq.
+ * \param nb_clusters The number of clusters contained in \c membership. Must
+ *    not be a \c NULL pointer.
+ * \param quality The quality of the partition, in terms of the objective
+ *    function as included in the documentation. If \c NULL the quality will
+ *    not be calculated.
+ * \return Error code.
+ *
+ * Time complexity: near linear on sparse graphs.
+ *
+ * \example examples/simple/igraph_community_leiden.c
+ */
+int igraph_community_leiden(const igraph_t *graph,
+  const igraph_vector_t *edge_weights, const igraph_vector_t *node_weights,
+  const igraph_real_t resolution_parameter, const igraph_real_t beta,
+  igraph_vector_t *membership, igraph_integer_t *nb_clusters, igraph_real_t *quality) 
+{
+  igraph_vector_t *i_edge_weights, *i_node_weights;
+  int ret;
+
+  if (igraph_vector_size(membership) != igraph_vcount(graph)) 
+    IGRAPH_ERROR("Membership length does not equal the number of vertices", IGRAPH_EINVAL);
+
+  if (igraph_is_directed(graph))
+    IGRAPH_ERROR("Leiden algorithm is only implemented for undirected graphs", IGRAPH_EINVAL);
+    
+  /* Check edge weights to possibly use default */
+  if (!edge_weights)
+  {
+      i_edge_weights = igraph_Calloc(1, igraph_vector_t);
+      if (i_edge_weights == 0)
+        IGRAPH_ERROR("Leiden algorithm failed, could not allocate memory for edge weights", IGRAPH_ENOMEM);    
+      IGRAPH_CHECK(igraph_vector_init(i_edge_weights, igraph_ecount(graph)));
+      IGRAPH_FINALLY(free, i_edge_weights);
+      IGRAPH_FINALLY(igraph_vector_destroy, i_edge_weights);
+      igraph_vector_fill(i_edge_weights, 1);
+  }
+  else
+      i_edge_weights = edge_weights;
+
+  /* Check edge weights to possibly use default */
+  if (!node_weights)
+  {
+      i_node_weights = igraph_Calloc(1, igraph_vector_t);
+      if (i_node_weights == 0)
+        IGRAPH_ERROR("Leiden algorithm failed, could not allocate memory for node weights", IGRAPH_ENOMEM);    
+      IGRAPH_CHECK(igraph_vector_init(i_node_weights, igraph_vcount(graph)));
+      IGRAPH_FINALLY(free, i_node_weights);
+      IGRAPH_FINALLY(igraph_vector_destroy, i_node_weights);
+      igraph_vector_fill(i_node_weights, 1);
+  }
+  else
+      i_node_weights = node_weights;
+
+  /* Perform actual Leiden algorithm */
+  ret = igraph_i_community_leiden(graph, i_edge_weights, i_node_weights,
+                                      resolution_parameter, beta,
+                                      membership, nb_clusters, quality);
+
+  if (!edge_weights)
+  {
+    igraph_vector_destroy(i_edge_weights);
+    igraph_Free(i_edge_weights);
+    IGRAPH_FINALLY_CLEAN(1);
```

Comment:
I think this should be `2` and not `1` (one for the `vector_destroy`, one for the `free`).